File: frontend/src/backend/shoppingService.js
import {appFetch} from './appFetch';

export const buyTickets = (sessionId, quantity, creditCard) =>
    appFetch('POST', '/shopping/purchases', {
        sessionId: Number(sessionId),
        quantity: Number(quantity),
        creditCard
    });

export const findPurchases = ({page}) =>
    appFetch('GET', `/shopping/purchases?page=${page}`);

export const deliverTickets = (purchaseId, creditCardNumber) =>
    appFetch('POST', `/shopping/purchases/${purchaseId}/deliver`, {
        creditCardNumber
    });

File: backend/src/main/java/es/udc/paproject/backend/model/services/ShoppingService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Purchase;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.NotEnoughSeatsException;
import es.udc.paproject.backend.model.exceptions.SessionAlreadyStartedException;
import es.udc.paproject.backend.model.exceptions.IncorrectPurchaseCreditCardException;
import es.udc.paproject.backend.model.exceptions.TicketsAlreadyDeliveredException;
import es.udc.paproject.backend.model.exceptions.MaxTicketsExceededException;

public interface ShoppingService {

    Purchase buyTickets(Long userId, Long sessionId, int tickets, String creditCard)
            throws InstanceNotFoundException, NotEnoughSeatsException, SessionAlreadyStartedException, MaxTicketsExceededException;

    Block<Purchase> findPurchases(Long userId, int page, int size);

    Purchase deliverTickets(Long purchaseId, String creditCard)
            throws InstanceNotFoundException, IncorrectPurchaseCreditCardException, TicketsAlreadyDeliveredException, SessionAlreadyStartedException;

}
